### ISSUE TITLE
Add new response type(s) ("HtmlResponse?") for responses that do not want header/footer but HTML rendering

### ISSUE THREAD

**Issue by MasterOdin** [Submitty/Submitty#6218]
Looking at #6215, the `getSingleGradeInquiryPost` works by disabling Header/Footer, and then returning either `""` or a WebResponse. This works because the returned `""` is technically a no-op (it could just do `return;` and still work) as on coming back from the method, we just enter the `Output` library, which does nothing (as no header / footer), and so just trivially returns. However, this relies on the fact you have to remember to turn off header/footer before returning anything, and can be a place that trips up a developer.

Instead, it would be better to add new response types (`HtmlResponse`, `TextResponse`?) to act as a shorthand of doing:

```
        $this->core->getOutput()->useHeader(false);
        $this->core->getOutput()->useFooter(false);
        return new WebResponse(
            ['submission', 'Homework'],
            'renderSingleGradeInquiryPost',
            $new_post,
            $graded_gradeable
        );
```

and instead would do just:
```
        return new HtmlResponse(
            ['submission', 'Homework'],
            'renderSingleGradeInquiryPost',
            $new_post,
            $graded_gradeable
        );
```

and also instead of returning `""`, we do `return TextResponse('');`. This allows us to keep consistency in controller return types (Responses), as well as reduce a bit of cognitive overhead of remembering to turn off header/footer and also what does `return ""` actually mean (nothing).

**Comment by shailpatels** [comment]
I like this idea, and yeah the `getSingleGradeInquiryPost` I threw ""s to get it finished quickly due to the type of bug its trying to fix. There are a couple of other places that depend on fetching HTML through ajax to dynamically update the page that an HTML response would be helpful in.

I can also see it being used in the ta grading interface which renders a lot through twig.js, getting everything serverside in pieces would be away to remove it

**Comment by MasterOdin** [comment]
Yeah, I think there are other places in the codebase that would benefit and probably other places in the codebase that follow a similar no-op paradigm, that one place was just the one I noticed, and that I don't necessarily think we need to spend time refactoring things in #6215 and get it merged sooner rather than later.
